daphne/src/roles_test.rs

```diff
@@ -357,10 +392,14 @@ impl Test {
         let req = self
             .collector_authorized_req(
                 task_config.version,
-                MEDIA_TYPE_COLLECT_REQ,
+                versioned_media_type_for(&task_config.version, MEDIA_TYPE_COLLECT_REQ).unwrap(),
                 task_id,
-                CollectReq {
-                    task_id: task_id.clone(),
+                CollectionReq {
+                    draft02_task_id: if task_config.version == DapVersion::Draft02 {
+                        Some(task_id.clone())
+                    } else {
+                        None
+                    },
```

Comment:
I'm pretty  sure this works (here and elsewhere)
```suggestion
                    draft02_task_id: task_id.for_request_payload(task_config.version),
```